cocos/ui/UIRichText.cpp

```diff
@@ -1795,19 +1834,28 @@ void RichText::formatRenderers()
             float maxHeight = 0.0f;
             for (auto& iter : row)
             {
-                maxHeight = MAX(iter->getContentSize().height, maxHeight);
+                maxHeight = std::max(iter->getContentSize().height, maxHeight);
+            }
+
+            // gap for empty line, if _lineHeights[i] == 0, use current RichText's fontSize
+            if (row.empty())
+            {
+                maxHeight = (_lineHeights[i] != 0.0f ? _lineHeights[i] : fontSize);
```

Comment:
Why use font size for height? And when row will be empty?

Reply:
Consider the following two cases：
- a `RichText` with fontSize 20 contains `RichElementText`(string="abc\n\n123", fontSize=10).
Then it will display like this:
```
abc

123
```
Now,  the gap between two lines should be equal to `RichElementText`'s fontSize 10

- a `RichText` with fontSize 20 contains `RichElementText`("abc"), `RichElementNewLine`, `RichElementNewLine`,  `RichElementText`("123").
Then it will display like this:
```
abc

123
```
Now,  the gap between two lines should be equal to `RichText`'s fontSize 20

In both of two cases, row.empty() == true. because of `RichText::addNewLine()`, the code `_elementRenders.emplace_back();`

Reply:
Why not just use vertical space instead of font size? Use font size as line height is strange.

Reply:
please see below

Reply:
sounds reasonable